test/integration/testPutGet.js

```diff
@@ -857,3 +861,315 @@ describe('PUT GET test multiple files', function () {
+
+  const testCases =
+    [
+      {
+        name: 'no compress'
+      },
+    ];
+
+  const createItCallback = function () {
+    return function (done) {
+      {
+        async.series(
+          [
+            function (callback) {
+              connection.execute({
+                sqlText: putQuery,
+                complete: function (err, stmt) {
+                  if(err){
+                    callback(err);
+                  }else {
+                    const stream = stmt.streamRows();
+                    stream.on('error', function (err) {
+                      callback(err);
+                    });
+                    stream.on('data', function (row) {
+                      // Check the file is correctly uploaded
+                      assert.strictEqual(row['status'], UPLOADED);
+                      assert.strictEqual(row.targetSize, ROW_DATA_SIZE);
+                    });
+                    stream.on('end', function () {
+                      callback();
+                    });
+                  }
+                }
+              });
+            },
+            function (callback) {
+              connection.execute({
+                sqlText: getQuery,
+                complete: function (err, stmt) {
+                  if(err){
+                    callback(err);
+                  }else {
+                    const stream = stmt.streamRows();
+                    stream.on('error', function (err) {
+                      callback(err);
+                    });
+                    stream.on('data', function (row) {
+                      assert.strictEqual(row['status'], DOWNLOADED);
+                      // Verify if the file is not compressed.
+                      const file = path.join(tmpDir, row.file);
+                      const data = fs.readFileSync(file).toString();
+                      assert.strictEqual(data, ROW_DATA);
+                    });
+                    stream.on('end', function () {
+                      callback();
+                    });
+                  }
+                }
+              });
+            }
+          ],
+          done
+        );
+      }
+
+    };
+  };
+
+  for (let index = 0; index < testCases.length; index++) {
+    const testCase = testCases[index];
+    it(testCase.name, createItCallback());
+  }
+});
+
+describe('PUT GET test with different size', function () {
+  let connection;
+  const stage = `@${DATABASE_NAME}.${SCHEMA_NAME}.%${TEMP_TABLE_NAME}`;
+  const createTable = `create or replace table ${TEMP_TABLE_NAME} (${COL1} STRING, ${COL2} STRING, ${COL3} STRING)`;
+  const removeFile = `REMOVE ${stage}`;
+  const dropTable = `DROP TABLE IF EXISTS ${TEMP_TABLE_NAME}`;
+
+  // Create a temp file without specified file extension
+  const zeroByteFile = tmp.fileSync();
+  const largeFile = tmp.fileSync(); 
+  // Create a tmp folder for downloaded files
+  const tmpDir = fs.mkdtempSync(path.join(os.tmpdir(), 'get'));
```

Comment:
let's create dir in before section

Reply:
updated.

Reply:
using let is not a problem - it's still better than var :)

the problem is that describe content is interpreted synchronously before even running tests so we are generating filesystem calls even when it's not necessary, example: 

1. I added some logs inside describe of `PUT GET test without compress` and `PUT GET test with different size`
2. Mark describe `PUT GET test without compress` with skip
3. Mark describe `PUT GET test with different size` with only
4. Run the test

Output:
```
$ npm run test:single -- test/integration/testPutGet.js 

> snowflake-sdk@1.6.23 test:single
> mocha -timeout 180000 --full-trace test/integration/testPutGet.js

Running describe of PUT GET test without compress
Running describe of PUT GET test with different size
(node:57491) NOTE: We are formalizing our plans to enter AWS SDK for JavaScript (v2) into maintenance mode in 2023.

Please migrate your code to use AWS SDK for JavaScript (v3).
For more information, check the migration guide at https://a.co/7PzMCcy
(Use `node --trace-warnings ...` to show where the warning was created)


  PUT GET test with different size
    ✔ zero byte file (4172ms)
    ✔ large size file (71411ms)


  2 passing (1m)
```

Reply:
@sfc-gh-dprzybysz you are right. Describe sections shouldn't contains operations making changes on fs.

Reply:
I've made the change. 
Only a small issue, I cannot use the testcases anymore. Because in this line **it(testCase.name, createItCallback(testCase))** testCase is pre-assigned/compiled before executing **before()**